x-pack/filebeat/module/microsoft/defender_atp/ingest/pipeline.yml

```diff
@@ -0,0 +1,327 @@
+- rename:
+    field: json.aadTenantId
+    target_field: cloud.account.id
+    ignore_missing: true
+- rename:
+    field: json.machineId
+    target_field: cloud.instance.id
+    ignore_missing: true
+- rename:
+    field: json.title
+    target_field: message
+    ignore_missing: true
+#######################
+## ECS Event Mapping ##
+#######################
+- set:
+    field: event.kind
+    value: alert
+# Events returned from the API is always in UTC, so should never use anything else
+- set:
+    field: event.timezone
+    value: UTC
+- set:
+    field: event.action
+    value: '{{json.category}}'
+    if: ctx.json?.category != null
+- set:
+    field: event.provider
+    value: defender_atp
+- set:
+    field: event.ingested
+    value: '{{_ingest.timestamp}}'
+- set:
+    field: event.created
+    value: '{{json.alertCreationTime}}'
+    if: ctx.json?.alertCreationTime != null
+- append:
+    field: event.category
+    value: host
+- append:
+    field: event.category
+    value: malware
+    if: ctx.json?.category == 'Malware'
+- append:
+    field: event.category
+    value: process
+    if: ctx.json?.evidence?.entityType == 'Process'
+- append:
+    field: event.type
+    value: user
+    if: ctx.json?.evidence?.entityType == 'User'
+- append:
+    field: event.type
+    value:
+      - creation
+      - start
+    if: ctx.json?.status == 'New'
+- append:
+    field: event.type
+    value: end
+    if: ctx.json?.status == 'Resolved'
+- rename:
+    field: json.id
+    target_field: event.id
+    ignore_missing: true
+- rename:
+    field: json.firstEventTime
+    target_field: event.start
+    ignore_missing: true
+- rename:
+    field: json.lastEventTime
+    target_field: event.end
+    ignore_missing: true
+- set:
+    field: event.severity
+    value: 0
+    if: ctx.json?.severity == 'Unspecified'
+- set:
+    field: event.severity
+    value: 1
+    if: ctx.json?.severity == 'Informational'
+- set:
+    field: event.severity
+    value: 2
+    if: ctx.json?.severity == 'Low'
+- set:
+    field: event.severity
+    value: 3
+    if: ctx.json?.severity == 'Medium'
+- set:
+    field: event.severity
+    value: 4
+    if: ctx.json?.severity == 'High'
+- script:
+    lang: painless
+    if: "ctx?.event?.start != null && ctx?.event?.end != null"
+    source: >
+      Instant eventstart = ZonedDateTime.parse(ctx?.event?.start).toInstant();
+      Instant eventend = ZonedDateTime.parse(ctx?.event?.end).toInstant();
+      ctx.event['duration'] = ChronoUnit.NANOS.between(eventstart, eventend);
+
+########################
+## ECS Threat Mapping ##
+########################
+- set:
+    field: threat.framework
+    value: MITRE ATT&CK
+    if: ctx.json?.category != null
+- rename:
+    field: json.category
+    target_field: threat.technique.name
+    ignore_missing: true
+- rename:
+    field: json.description
+    target_field: rule.description
+    ignore_missing: true
+    if: (ctx.json?.description).length() < 1020
```

Comment:
I think that the sentence would just break off at that point, it would be half a sentence. I think that its just to ensure it never hits the limit as it was the only field that can in very niche usecases maybe include too much info. It wouldn't stop any of the events I have seen, and its more to drop it on ingest rather than creating an error in elasticsearch due to the field size limit (1024?)